src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -0,0 +1,681 @@
+
+            QUIC_PACKET_KEY* WriteKey;
+            
+            if (QUIC_FAILED(
+                QuicPacketKeyCreateInitial(
+                    FALSE,
+                    &HkdfLabels,
+                    InitialSalt.Data,
+                    sizeof(uint64_t),
+                    (uint8_t*)DestCid,
+                 nullptr,
+                    &WriteKey))) {
+                        printf("QuicPacketKeyCreateInitial failed\n");
+                        return;
+                }
+
+            // printf_buf("salt", InitialSalt.Data, InitialSalt.Length);
+            // printf_buf("cid", DestCid, sizeof(uint64_t));
+
+            uint8_t Iv[CXPLAT_IV_LENGTH];
+            QuicCryptoCombineIvAndPacketNumber(
+                WriteKey->Iv, (uint8_t*)&PacketNumber, Iv);
+
+            CxPlatEncrypt(
+                WriteKey->PacketKey,
+                Iv,
+                HeaderLength,
+                SendBuffer->Buffer,
+                PacketLength - HeaderLength,
+                SendBuffer->Buffer + HeaderLength);
+
+            // printf_buf("encrypted", SendBuffer->Buffer, PacketLength);
+
+            uint8_t HpMask[16];
+            CxPlatHpComputeMask(
+                WriteKey->HeaderKey,
+                1,
+                SendBuffer->Buffer + HeaderLength,
+                HpMask);
+
+            // printf_buf("cipher_text", SendBuffer->Buffer + HeaderLength, 16);
+            // printf_buf("hp_mask", HpMask, 16);
+            QuicPacketKeyFree(WriteKey);
+
+            SendBuffer->Buffer[0] ^= HpMask[0] & 0x0F;
+            for (uint8_t i = 0; i < 4; ++i) {
+                SendBuffer->Buffer[PacketNumberOffset + i] ^= HpMask[i + 1];
+            }
+            
+            // printf_buf("protected", SendBuffer->Buffer, PacketLength);
+            InterlockedExchangeAdd64(&PacketCount, 1);
+            InterlockedExchangeAdd64(&TotalByteCount, DatagramLength);
+
+        }
+
+        
+        QUIC_SUCCEEDED(
+        CxPlatSocketSend(
+            Binding,
+            &Route,
+            SendData));
+    }
+    printf("Total Initial Packets sent: %lld\n", (long long)PacketCount);
+    printf("Total Bytes sent: %lld\n", (long long)TotalByteCount);
+}
+
+void start() {
+    CXPLAT_DATAPATH* Datapath;
+    const CXPLAT_UDP_DATAPATH_CALLBACKS DatapathCallbacks = {
+        UdpRecvCallback,
+        UdpUnreachCallback,
+    };
+    MsQuic = new MsQuicApi();
+    QUIC_STATUS Status = CxPlatDataPathInitialize(
+        0,
+        &DatapathCallbacks,
+        NULL,
+        NULL,
+        &Datapath);
+    if (QUIC_FAILED(Status)) {
+        printf("Datapath init failed 0x%x", Status);
+        return;
+    }
+    QUIC_ADDR sockAddr = {0};
+    auto value = GetRandom(3);
+    QUIC_ADDRESS_FAMILY Family = (value == 0) ? QUIC_ADDRESS_FAMILY_INET6 : ((value == 1) ? QUIC_ADDRESS_FAMILY_INET : QUIC_ADDRESS_FAMILY_UNSPEC); // fuzz
```

Comment:
Instead of using `3` and testing `UNSPEC`, just do `2` and do v4 or v6.
```suggestion
    QUIC_ADDRESS_FAMILY Family = GetRandom(2) ? QUIC_ADDRESS_FAMILY_INET6 : QUIC_ADDRESS_FAMILY_INET; // fuzz
```